src/redux/uniswap.ts

```diff
@@ -155,6 +182,46 @@ export const uniswapResetState = () => (
   dispatch: Dispatch<UniswapClearStateAction>
 ) => dispatch({ type: UNISWAP_CLEAR_STATE });
 
+/**
+ * Loads uniswap favorites metadata from local storage or fetches new data
+ * on update / when persisting default favorites for the first time
+ */
+const getUniswapFavoritesMetadata = async (
+  addresses: EthereumAddress[],
+  network?: string
+) => {
+  let favoritesMetadata = await getUniswapFavoritesMetadataLS(network);
+  if (Object.keys(favoritesMetadata).length !== addresses.length) {
```

Comment:
`favoritesMetadata` could be undefined (say for other networks) - running `Object.keys` on it throws an error